manim/utils/color.py

```diff
@@ -1,15 +1,135 @@
 from colour import Color
 import numpy as np
 
-from ..constants import COLOR_MAP
-from ..constants import WHITE
 from ..utils.bezier import interpolate
 from ..utils.simple_functions import clip_in_place
 from ..utils.space_ops import normalize
 
 
+class Colors(Enum):
+    """A list of pre-defined colors.
+
+    Examples
+    --------
+
+    The preferred way of using these colors is
+
+    .. code-block:: python
+
+       import manim.utils.color as C
+       C.WHITE   # -> '#FFFFFF'
+
+    Note this way uses the name of the colors below in UPPERCASE.
+
+    Alternatively, you can also import this Enum directly and use its members
+    directly, through the use of :code:`color.value`.  Note this way uses the
+    name of the colors in lowercase.
+
+    .. code-block:: python
+
+       from manim.utils.color import Colors
+       Colors.white.value   # -> '#FFFFFF'
+
+    """
+
+    dark_blue = "#236B8E"
+    dark_brown = "#8B4513"
+    light_brown = "#CD853F"
+    blue_e = "#1C758A"
+    blue_d = "#29ABCA"
+    blue_c = "#58C4DD"
+    blue = "#58C4DD"
+    blue_b = "#9CDCEB"
+    blue_a = "#C7E9F1"
+    teal_e = "#49A88F"
+    teal_d = "#55C1A7"
+    teal_c = "#5CD0B3"
+    teal = "#5CD0B3"
+    teal_b = "#76DDC0"
+    teal_a = "#ACEAD7"
+    green_e = "#699C52"
+    green_d = "#77B05D"
+    green_c = "#83C167"
+    green = "#83C167"
+    green_b = "#A6CF8C"
+    green_a = "#C9E2AE"
+    yellow_e = "#E8C11C"
+    yellow_d = "#F4D345"
+    yellow_c = "#FFFF00"
+    yellow = "#FFFF00"
+    yellow_b = "#FFEA94"
+    yellow_a = "#FFF1B6"
+    gold_e = "#C78D46"
+    gold_d = "#E1A158"
+    gold_c = "#F0AC5F"
+    gold = "#F0AC5F"
+    gold_b = "#F9B775"
+    gold_a = "#F7C797"
+    red_e = "#CF5044"
+    red_d = "#E65A4C"
+    red_c = "#FC6255"
+    red = "#FC6255"
+    red_b = "#FF8080"
+    red_a = "#F7A1A3"
+    maroon_e = "#94424F"
+    maroon_d = "#A24D61"
+    maroon_c = "#C55F73"
+    maroon = "#C55F73"
+    maroon_b = "#EC92AB"
+    maroon_a = "#ECABC1"
+    purple_e = "#644172"
+    purple_d = "#715582"
+    purple_c = "#9A72AC"
+    purple = "#9A72AC"
+    purple_b = "#B189C6"
+    purple_a = "#CAA3E8"
+    white = "#FFFFFF"
+    black = "#000000"
+    light_gray = "#BBBBBB"
+    light_grey = "#BBBBBB"
+    gray = "#888888"
+    grey = "#888888"
+    dark_grey = "#444444"
+    dark_gray = "#444444"
+    darker_grey = "#222222"
+    darker_gray = "#222222"
+    grey_brown = "#736357"
+    pink = "#D147BD"
+    light_pink = "#DC75CD"
+    green_screen = "#00FF00"
+    orange = "#FF862F"
+
+
+# Having a list of pre-defined colors is useful for auto-completion in several
+# IDEs.  However, in order to reach the value of a color using the Colors enum,
+# one needs to access Colors.white.value, which is far too long and clunky.
+# Therefore, we also add each color as a global constant, in upper case, to the
+# locals() of this module
+COLOR_MAP = {n.upper(): c.value for n, c in Colors.__members__.items()}
+locals().update(COLOR_MAP)
```

Comment:
With this approach,  unfortunately, we don't get auto-completion

Reply:
Oh no! I didn't realize that. Thanks for catching that